build-tests/api-documenter-test/etc/api-documenter-test.api.json

```diff
@@ -59,6 +63,7 @@
           "members": [
             {
               "kind": "Constructor",
+              "canonicalReference": "api-documenter-test!DocBaseClass#:constructor(1)",
```

Comment:
@rbuckton is this the expected UID for a constructor?

Reply:
No. I'd prefer to see `api-documenter-test!DocBaseClass:constructor(1)`. In TypeScript, constructors have a reserved internal name of `__constructor`, but `api-documenter-test!DocBaseClass#__constructor:constructor(1)` seems unnecessarily repetitive.

Reply:
> No. I'd prefer to see `api-documenter-test!DocBaseClass:constructor(1)`.

@rbuckton  This string can be parsed successfully, but how do we it using your API?

Reply:
Oh I see, it's  like this:
```ts
DeclarationReference.empty()
  .addNavigationStep(".", "DocBaseClass")
  .withMeaning("constructor")
  .withOverloadIndex(1);
```
I'll update the PR.

Reply:
I pushed an update that fixes all the unnamed types; lemme know if that looks better.